src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -4886,6 +5009,85 @@ export class StudyViewPageStore
         return this.genomicDataCountPromises[chartMeta.uniqueKey];
     }
 
+    public getMutationEventChartDataCount(
+        chartMeta: ChartMeta
+    ): MobxPromise<MultiSelectionTableRow[]> {
+        if (
+            !this.mutationDataCountPromises.hasOwnProperty(chartMeta.uniqueKey)
+        ) {
+            this.mutationDataCountPromises[chartMeta.uniqueKey] = remoteData<
+                MultiSelectionTableRow[]
+            >({
+                await: () => [this.selectedSamples],
+                invoke: async () => {
+                    let res: MultiSelectionTableRow[] = [];
+                    const chartInfo = this._geneSpecificChartMap.get(
+                        chartMeta.uniqueKey
+                    );
+                    //only invoke if there are filtered samples
+                    if (chartInfo && this.hasFilteredSamples) {
+                        const params = {
+                            genomicDataCountFilter: {
+                                genomicDataFilters: [
+                                    {
+                                        hugoGeneSymbol:
+                                            chartInfo.hugoGeneSymbol,
+                                        profileType: chartInfo.profileType,
+                                    },
+                                ] as any,
+                                studyViewFilter: this.filters,
+                            },
+                            $queryParameters: {
+                                projection:
+                                    chartInfo.mutationOptionType ===
+                                    MutationOptionConstants.MUTATED
+                                        ? 'SUMMARY'
+                                        : 'DETAILED',
+                            },
+                        } as any;
+
+                        const [result, selectedSamples] = await Promise.all([
+                            await internalClient.fetchMutationDataCountsUsingPOST(
+                                params
+                            ),
+                            toPromise(this.selectedSamples),
```

Comment:
selectedSamples is awaited in by the mobxpromise, so you shouldn't need to handle this here.  you can just use it.

Reply:
removed